src/hash.rs

```diff
@@ -64,3 +74,51 @@ fn file_hash(algorithm: &str, path: &str) -> Result<String> {
 
     Ok(hash_algorithm(algorithm, &bytes)?)
 }
+
+fn totp_generate_tolerance(hex_seed: &str, tolerance: i32) -> Result<String> {
+    let mut result: String = "".to_string();
+    for i in -tolerance..(tolerance + 1) {
+        if i != -tolerance {
+            result.push(',');
+        }
+        result += &totp_generate(hex_seed, i.try_into().unwrap()).unwrap();
+        println!("Result is now {}", result)
+    }
+    Ok(result.to_string())
+}
+
+fn totp_generate(hex_seed: &str, offset: i64) -> Result<String> {
+    let mut seed: [u8; 64] = [0; 64];
+    let mut seed_bytes = [0u8; 10];
+
+    hex::decode_to_slice(hex_seed, &mut seed_bytes as &mut [u8]).unwrap();
```

Comment:
Can't this decode directly into `seed`? I'm not seeing the reason to have both seed and seed_bytes.

Reply:
Seed is too long, decode_to_slice panics due to incorrect string length

Reply:
Could you pass a sub-slice of seed that is the correct length?

Reply:
That worked